### ISSUE TITLE
Integration expects IP but UNVR has domain

### ISSUE THREAD

**Issue by VederJay** [briis/unifiprotect#436]
### Home Assistant Version?

core-2021.12.6

### UniFi Protect Version?

1.20.3

### UniFi Protect HA Integration version?

0.11.4 - also tried 0.12.0-beta3

### What happened?

Everything works up to the screen where the server address and the credentials are entered. After submission of these settings, I get "Success! Created configuration for UNVR myUNVR." 
When clicking "Finish", the integration appears but has a warning "Failed to set up". In the logs, the below appears. Apparently some part of the integration expects to get an IP and got a domain instead (I replaced the actual domain value with unvr.mydomain.com). 

### Relevant log output

```shell
Logger: homeassistant.config_entries
Source: .local/lib/python3.9/site-packages/pyunifiprotect/utils.py:163
First occurred: 20:54:04 (1 occurrences)
Last logged: 20:54:04

Error setting up entry UNVR myUNVR for unifiprotect
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/homeassistant/config_entries.py", line 313, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/unifiprotect/__init__.py", line 210, in async_setup_entry
    await data_service.async_setup()
  File "/config/custom_components/unifiprotect/data.py", line 74, in async_setup
    await self.async_refresh()
  File "/config/custom_components/unifiprotect/data.py", line 89, in async_refresh
    self._async_process_updates(await self.api.update(force=force))
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/api.py", line 568, in update
    self._bootstrap = await self.get_bootstrap()
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/api.py", line 717, in get_bootstrap
    return Bootstrap.from_unifi_dict(**data, api=self)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 100, in from_unifi_dict
    data = cls.unifi_dict_to_dict(data)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/bootstrap.py", line 132, in unifi_dict_to_dict
    return super().unifi_dict_to_dict(data)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 332, in unifi_dict_to_dict
    data[key] = cls._clean_protect_obj_dict(data[key], unifi_dicts[key], api)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 277, in _clean_protect_obj_dict
    items[key] = cls._clean_protect_obj(value, klass, api)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 261, in _clean_protect_obj
    return klass.unifi_dict_to_dict(data=data)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/devices.py", line 680, in unifi_dict_to_dict
    return super().unifi_dict_to_dict(data)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 595, in unifi_dict_to_dict
    return super().unifi_dict_to_dict(data)
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/data/base.py", line 315, in unifi_dict_to_dict
    data[key] = convert_unifi_data(data[key], cls.__fields__[key])
  File "/config/.local/lib/python3.9/site-packages/pyunifiprotect/utils.py", line 163, in convert_unifi_data
    value = IPv4Address(value)
  File "/usr/lib/python3.9/ipaddress.py", line 1307, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "/usr/lib/python3.9/ipaddress.py", line 1194, in _ip_int_from_string
    raise AddressValueError("Expected 4 octets in %r" % ip_str)
ipaddress.AddressValueError: Expected 4 octets in 'unvr.mydomain.com'
```

**Comment by AngellusMortis** [comment]
Duplicate of https://github.com/briis/unifiprotect/issues/432.

You can go back into your cameras and use IP addresses as w work around. I can try to get a new 0.12 beta out with the fix today if I can.

**Comment by AngellusMortis** [comment]
Sorry, wrong issue. https://github.com/briis/unifiprotect/issues/431

**Comment by VederJay** [comment]
Sorry, should have seen that already opened ticket. 

I too have some remote cameras connected to the UNVR. As I don't have a fixed IP address, I'm using a dynamic DNS service to keep my domain pointed towards my WAN IP, hence the need to use a domain instead of an IP. 
So the workaround isn't feasible as it would break my setup. 

Still, thanks for thinking along and thank you for any work you put into updating the integration. I realise it's all unpaid work and I can only say that it's much appreciated.

**Comment by AngellusMortis** [comment]
Released fix for other issue in 0.12-beta4. Just in case you did not follow that issue.
